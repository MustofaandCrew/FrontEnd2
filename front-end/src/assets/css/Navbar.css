/* NavBar */
.container-nav {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }
  
  nav {
    display: flex;
    width: 100%;
    background: #fff;
    position: relative;
    justify-content: space-between; 
    text-align: center;
    align-items: center;
    padding: 15px 30px;
    margin-bottom: 2%;
  }
  
  nav .icon {
    font-size: 35px;
    font-weight: 800;
    cursor: pointer;
  }
  
  nav .icon a{
    text-decoration: none;
    color: #000;
  }
  
  nav ol {
    display: flex;
    list-style: none;
    margin: auto 0;
  }
  
  nav ol li {
    margin: 0 2px;
  }
  
  nav ol li a {
    color: black;
    font-size: 20px;
    font-weight: 600;
    text-decoration: none;
    text-transform: capitalize;
    letter-spacing: 1px;
    padding: 5px 10px;
    cursor: pointer;
  }

  nav ol li p {
    color: black;
    font-size: 20px;
    font-weight: 600;
    text-decoration: none;
    text-transform: capitalize;
    letter-spacing: 1px;
    padding: 5px 10px;
    cursor: pointer;
  }
  
  nav ol li a:hover {
    border-radius: 8px;
    background: #FF7158;
    color: #fff;
    padding-bottom: 7px;
    transition: 0.5s;
  }

  nav ol li a.btn-seller {
    border-radius: 8px;
    background: #FF7158;
    color: #fff;
    padding: 10px;
    transition: 0.5s;
  }

  nav ol li a.btn-seller:hover {
    background: #ffab9c;
    color: #000;
  }

  nav ol li a.btn-seller img {
    width: 2rem;
    margin-bottom: 5px;
  }

  nav ol li a.profile-info {
    background: #fff;
    color: black;
    padding: 10px;
    transition: 0.5s;
  }

  nav ol li a.profile-info:hover {
    border-radius: 8px;
    background: rgb(222, 222, 222);
    color: black;
    padding: 10px;
    transition: 0.5s;
  }

  nav ol li a.profile-info img {
    width: 2rem;
    margin-bottom: 5px;
    margin-right: 5px;
  }
  
  nav ol.vertical-screen {
    display: none;
    list-style: none;
    margin: auto 0;
  }
  
  nav .search-box {
    display: flex;
    margin: auto 0;
    height: 35px;
    line-height: 35px;
  }
  
  nav .search-box input {
    border: none;
    outline: none;
    background: #EFEFEF;
    height: 100%;
    padding: 0 10px;
    font-size: 15px;
    width: 350px;
  }
  
  nav .search-box span {
    color: black;
    font-size: 20px;
    background: #EFEFEF;
    height: 100%;
    padding: 8px;
    position: relative;
    cursor: pointer;
    z-index: 1;
  }
  
  nav .search-box span:hover {
    color: #fff;
  }
  
  nav .search-box span::after {
    height: 100%;
    width: 0%;
    content: "";
    background: #FF7158;
    position: absolute;
    top: 0;
    right: 0;
    z-index: -1;
    transition: 0.3s;
  }
  
  nav .search-box span:hover::after {
    width: 100%;
  }
  
  nav .bar {
    position: relative;
    margin: auto;
    display: none;
  }
  
  nav .bar span {
    position: absolute;
    color: #fff;
    font-size: 35px;
  }
  
  input[type="checkbox"] {
    -webkit-appearance: none;
    display: none;
  }
  
  .btn-dropdown {
    border: none;
    background: #fff;
  }
  
  /* Extra small devices (phones, 600px and down) */
  @media only screen and (max-width: 600px){
    /* ini untuk tampilan handphone */
    .home-title {
      text-align: center;
    }
  
    nav {
      display: flex;
      width: 100%;
      background: #fff;
      position: relative;
      justify-content: space-between; 
      text-align: left;
      align-items: center;
      padding: 15px 30px;
      margin-bottom: 2%;
    }
  
    nav .search-box {
      padding-left: 4%;
      padding-right: 4%;
      width: 100%;
      display: inline-flex;
      justify-content: center;
      margin-bottom: 15px;
    }
    nav .search-box input {
      width: 20px;
      background: #EFEFEF;
    }
  
    nav .icon {
      font-size: 20px;
      font-weight: 800;
      cursor: pointer;
    }
    
    nav .icon a{
      text-decoration: none;
      color: #000;
    }
  }
  
  
  @media screen and (max-width: 1250px) {
    nav {
      background: #fff;
      display: block;
      padding: 0;
    }
    nav .icon {
      display: inline-block;
      padding: 15px 30px;
    }
    nav .search-box {
      width: 100%;
      display: inline-flex;
      justify-content: center;
      margin-bottom: 15px;
    }
    nav .search-box input {
      width: 90%;
      background: #EFEFEF;
    }
    nav ol {
      display: none;
    }
    nav ol.vertical-screen {
      padding-left: 0;
      display: flex;
      flex-direction: column;
      background: #fff;
      height: 0;
      visibility: hidden;
      transition: 0.3s;
    }
    nav ol.vertical-screen li {
      text-align: left;
      transition: 0.3s 0.1s all;
      opacity: 0;
      margin: 0 0;
    }
    nav ol.vertical-screen li.nav-span{
      padding-left: 25px;
    }
    nav ol.vertical-screen li.nav-span a{
      background: none;
      color: black;
    }
    nav ol.vertical-screen li.nav-span div.dropup a:hover{
      background: none;
    }
    nav ol.vertical-screen li.nav-span div.dropup a{
      background: none;
      color: black;
    }
    nav ol.vertical-screen li.nav-span div.dropup a:hover{
      background: none;
      color: black;
    }
    nav ol.vertical-screen li.nav-signin {
      margin-top: 15px;
      text-align: center;
    }
    nav ol.vertical-screen li.nav-signin a span {
      padding: 10px 20px;
      border-radius: 10px;
      background:#FF7158;
    }
    nav ol.vertical-screen li.nav-signin a span.profile {
      padding: 10px 20px;
      border-radius: 10px;
      background: rgb(222, 222, 222);
    }
    nav ol.vertical-screen li.nav-signin a span img {
      width: 2rem;
      padding-bottom: 4px;
    }
    nav ol.vertical-screen li.nav-sigin a:hover {
      color: #fff;
      transition: 0.5s;
    }
    nav ol.vertical-screen li a:hover {
      color: #FF7158;
      transition: 0.5s;
    }
    nav ol.vertical-screen li a {
      font-size: 15px;
      padding: 12px 0;
      display: block;
    }
    nav ol.vertical-screen li p {
      font-size: 15px;
      padding: 12px 0 10px 0;
      margin-bottom: 0;
      display: block;
    }
    nav ol.vertical-screen li button {
      margin-left: 0;
      padding-left: 0;
    }
    .accordion-body a p {
     text-align: left;
     font-size: 10px;
    }
    nav ol.vertical-screen li:nth-child(1) {
      transform: translateX(-150px);
    }
    nav ol.vertical-screen li:nth-child(2) {
      transform: translateX(-200px);
    }
    nav ol.vertical-screen li:nth-child(3) {
      transform: translateX(-250px);
    }
    nav ol.vertical-screen li:nth-child(4) {
      transform: translateX(-300px);
    }
    nav ol.vertical-screen li:nth-child(5) {
      transform: translateX(-350px);
    }
    nav .bar {
      display: block;
      position: absolute;
      top: 20px;
      right: 60px;
      bottom: 105px;
      cursor: pointer;
    }
    nav .bar span {
      position: absolute;
      color: #000000;
      font-size: 30px;
    }
    nav .bar #times {
      display: none;
    }
    #check:checked ~ nav .bar #times {
      display: block;
    }
    #check:checked ~ nav .bar #bars {
      display: none;
    }
    #check:checked ~ nav ol {
      visibility: visible;
      height: 280px;
    }
    #check:checked ~ nav ol li:nth-child(1),
    #check:checked ~ nav ol li:nth-child(2),
    #check:checked ~ nav ol li:nth-child(3),
    #check:checked ~ nav ol li:nth-child(4),
    #check:checked ~ nav ol li:nth-child(5) {
      transform: translateX(0);
      opacity: 1;
    }
  }