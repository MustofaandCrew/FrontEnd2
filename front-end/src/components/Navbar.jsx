import React from "react";
import Modal from "./ModalNavbar";
import Notif from "../components/Notification";
import logo from "../assets/images/logo.svg";
import LogoSeller from "../assets/images/logo-seller.svg";

// dummy images
import pp from "../assets/images/comments-pp.png";

import "../assets/css/Navbar.css";

export default function Navbar() {
    return (
        <div className="container-nav">
            <Modal />
            <input type="checkbox" id="check" />
            <nav>
                <div className="icon"><a href="/">SecondHand</a></div>
                <ol>
                    <li><a href="/">Home</a></li>
                    <div className="dropdown">
                        <li>
                            <button className="btn-dropdown" type="button" id="dropdownMenuButton1" data-bs-toggle="dropdown" aria-expanded="false">
                                <a href="/">Categories <i className="bx bx-chevron-down"></i></a>
                            </button>
                            <ul className="dropdown-menu" aria-labelledby="dropdownMenuButton1">
                                <li><a className="dropdown-item" href="/">Gadgets</a></li>
                                <li><a className="dropdown-item" href="/">Furniture</a></li>
                                <li><a className="dropdown-item" href="/">Make Up</a></li>
                                <li><a className="dropdown-item" href="/">Sneakers</a></li>
                                <li><a className="dropdown-item" href="/">Tools</a></li>
                                <li><a className="dropdown-item" href="/">Baby</a></li>
                            </ul>
                        </li>
                    </div>
                </ol>
                <div className="search-box">
                    <input type="search" placeholder="Search..." />
                    <span className="fa fa-search"></span>
                </div>
                <ol>
                    <li><a href="/"><span className="fa fa-shopping-basket"></span></a></li>
                    <li><a href="/"><span className="fa fa-bell"></span></a></li>
                    <li><h4>|</h4></li>
                    {/* <li><a href="/login">Sign In</a></li> */}
                    <li><a className="btn-seller" href="/"><img src={LogoSeller} alt="seller" /> Seller</a></li>
                    <li><a className="profile-info" href="/"><img className="img-fluid rounded-circle" src={pp} alt="profile" />Hi, Angga</a></li>
                </ol>
                <ol className="vertical-screen">
                    <li className="nav-span"><a href="/">Home</a></li>

                    <li className="nav-span">
                        <button type="button" className="btn-dropdown" data-bs-toggle="modal" data-bs-target="#exampleModal">
                            <p>Categories</p>
                        </button>
                    </li>

                    <li className="nav-span"><a href="/shoppingcart">Shopping Cart</a></li>
                    <li className="nav-span"><a href="/notification">Notification</a></li>
                    {/* <li className="nav-signin"><a href="/login"><span><i className="bx bx-log-in"></i> Sign In</span></a></li> */}
                    <li className="d-flex justify-content-center gap-2 nav-signin">
                        <a href="/"><span><img className="img-fluid" src={LogoSeller} alt="seller" /></span></a>
                        <a href="/"><span className="profile"><img className="img-fluid rounded-circle" src={pp} alt="profile" /></span></a>
                    </li>
                </ol>
                <label for="check" className="bar">
                    <span className="fa fa-bars" id="bars"></span>
                    <span className="fa fa-times" id="times"></span>
                </label>
            </nav>
        </div>
    )
}

