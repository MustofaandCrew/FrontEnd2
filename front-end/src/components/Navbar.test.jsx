import React from "react";
import renderer from "react-test-renderer";
import Navbar from "../components/Navbar";


describe("Test suits for Navbar", () => {
    it("should match with snapshot", () => {
        const tree = renderer.create(<Navbar />).toJSON();
    });
});