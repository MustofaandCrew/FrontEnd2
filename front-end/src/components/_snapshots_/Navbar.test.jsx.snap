// Jest Snapshot v1, https://goo.gl/fbAQLP
import Navbar from "../components/Navbar";

exports["Navbar"] = {
    test: function test() {
        var tree = renderer.create(<Navbar />).toJSON();
        expect(tree).toMatchSnapshot();
    }
};